File: leet/reverse_linked_list.rb
class ListNode
    attr_accessor :val, :next
    def initialize(val = 0, _next = nil)
        @val = val
        @next = _next
    end
end

def reverse_list(head)
    cur = head
    minus = nil
    while cur != nil
        plus = cur.next
        cur.next = minus
        minus = cur
        cur = plus
    end
    return minus
end

reverse_list([1,2,3,4,5])

File: python/reverse_linked_list.py
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def reverseList(head):
        cur = head # 1
        minus = None
        while cur:
            plus = cur.next
            cur.next = minus # nil
            minus = cur # 1
            cur = plus # 2

        return minus

Solution.reverseList([1,2,3,4,5])